test/parallel/test-require-deps-deprecation.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+// The v8 modules when imported leak globals. Disable global check.
+common.globalCheck = false;
+
+const deprecatedModules = [
+  'node-inspect/lib/_inspect',
+  'node-inspect/lib/internal/inspect_client',
+  'node-inspect/lib/internal/inspect_repl',
+  'v8/tools/SourceMap',
+  'v8/tools/codemap',
+  'v8/tools/consarray',
+  'v8/tools/csvparser',
+  'v8/tools/logreader',
+  'v8/tools/profile',
+  'v8/tools/profile_view',
+  'v8/tools/splaytree',
+  'v8/tools/tickprocessor',
+  'v8/tools/tickprocessor-driver'
+];
+
+// Newly added deps that do not have a deprecation wrapper around it would
+// throw an error, but no warning would be emitted.
+const deps = [
+  'acorn/dist/acorn',
+  'acorn/dist/walk'
+];
+
+const throwingModules = () => {
+  for (const m of deprecatedModules) {
+    require(m);
+  }
+};
+
+assert.throws(throwingModules, ReferenceError);
```

Comment:
This doesn't seem correct, it should be inside the the loop or it will only test the first `require()` call that throws.

Reply:
True, but that's one of the problem: I'm not sure every require throws.

From what I've seen, some tools throw (typically `v8/tools/profile`), some other don't. Should I remove every module not throwing? Or is there a better way to test this?

Reply:
I'm not sure either, let's ping the reviewers of the original PR.
cc: @TimothyGu @cjihrig @Trott @jasnell

Reply:
The checking for `ReferenceError` appears to only check that `require(m)` is the wrong way to include two of the modules in the list. The correct way to include (for example) `'v8/tools/tickprocessor'` is not `require('v8/tools/tickprocessor')` but rather something like `eval(process.binding('natives')['v8/tools/tickprocessor'])` when run with `--expose-internals`.

I think the test for `ReferenceError` should be removed entirely. The important thing to test is that a DeprecationWarning is being emitted, using `common.expectsWarning()` or whatever it is that's in the `common` module for that. I'm going to change my review to Request Changes for that. Sorry that I missed that on first review!

Reply:
Oh, wait, I didn't approve it or even review this until now. It's only 8 hours old. Never mind. But still, the comment stands: Yeah, that should be changed. Sorry if I'm directly or indirectly responsible for that code in the first place.

Reply:
@Trott this is an attempt to backport #17848.

Reply:
@lpinca Thanks. I don't see a `ReferenceError` check in this file on the master branch so now I'm even more confused. :-D

Reply:
I'm not sure this should actually be backported.